SSLMockServerTest is flaky

```
[INFO] Running com.google.cloud.spanner.pgadapter.SSLMockServerTest
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.035 s <<< FAILURE! -- in com.google.cloud.spanner.pgadapter.SSLMockServerTest
Error:  com.google.cloud.spanner.pgadapter.SSLMockServerTest -- Time elapsed: 0.008 s <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.google.cloud.spanner.pgadapter.SSLMockServerTest.generateSSLKey(SSLMockServerTest.java:104)
	at com.google.cloud.spanner.pgadapter.SSLMockServerTest.startMockSpannerAndPgAdapterServers(SSLMockServerTest.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:49)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:120)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:95)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:69)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
```

--- a/src/test/java/com/google/cloud/spanner/pgadapter/SSLMockServerTest.java
+++ b/src/test/java/com/google/cloud/spanner/pgadapter/SSLMockServerTest.java
@@ -14,9 +14,11 @@
 
 package com.google.cloud.spanner.pgadapter;
 
+import static com.google.cloud.spanner.pgadapter.nodejs.NodeJSTest.readAll;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
 import static org.junit.Assume.assumeTrue;
 
 import com.google.cloud.spanner.MockSpannerServiceImpl;
@@ -25,8 +27,11 @@ import com.google.cloud.spanner.pgadapter.wireprotocol.SSLMessage;
 import io.opentelemetry.api.OpenTelemetry;
 import java.io.BufferedReader;
 import java.io.File;
+import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.net.InetAddress;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
 import java.util.stream.Collectors;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
@@ -44,6 +49,10 @@ public class SSLMockServerTest extends AbstractMockServerTest {
   public static void startMockSpannerAndPgAdapterServers() throws Exception {
     assumeTrue("This test requires psql to be installed", isPsqlAvailable());
     assumeTrue("This test requires keytool to be installed", isKeytoolAvailable());
+    assumeTrue("This test requires an RSA key generator", isKeyPairGeneratorAvailable("RSA"));
+    assumeFalse(
+        "Host names containing an underscore are not allowed",
+        InetAddress.getLocalHost().getHostName().contains("_"));
 
     String keystore = generateSSLKey();
     publicKeyFile = exportPublicKey(keystore);
@@ -57,6 +66,15 @@ public class SSLMockServerTest extends AbstractMockServerTest {
         "d",
         builder -> builder.setSslMode(SslMode.Require),
         OpenTelemetry.noop());
+  }
+
+  private static boolean isKeyPairGeneratorAvailable(String generator) {
+    try {
+      KeyPairGenerator.getInstance(generator);
+    } catch (NoSuchAlgorithmException ignore) {
+      return false;
+    }
+    return true;
   }
 
   @AfterClass
@@ -99,9 +117,24 @@ public class SSLMockServerTest extends AbstractMockServerTest {
         };
     keytoolBuilder.command(keytoolCommand);
     Process keytoolProcess = keytoolBuilder.start();
+    InputStream outputStream = keytoolProcess.getInputStream();
+    InputStream errorStream = keytoolProcess.getErrorStream();
 
     int res = keytoolProcess.waitFor();
-    assertEquals(0, res);
+    String output = readAll(outputStream);
+    String errors = readAll(errorStream);
+    assertEquals(
+        "INFO:\n"
+            + output
+            + "\n\nERROR:\n"
+            + errors
+            + "\n\nHOST: "
+            + InetAddress.getLocalHost().getHostName()
+            + " ("
+            + InetAddress.getLocalHost().getHostAddress()
+            + ")",
+        0,
+        res);
 
     return keystore.getAbsolutePath();
   }
@@ -264,18 +297,20 @@ public class SSLMockServerTest extends AbstractMockServerTest {
   @Test
   public void testSSLVerifyFull() throws Exception {
     for (String host :
-        new String[] {
-          InetAddress.getLocalHost().getHostName(),
-          InetAddress.getLocalHost().getHostAddress(),
-          "localhost"
-        }) {
+        System.getProperty("os.name").contains("Mac")
+            ? new String[] {InetAddress.getLocalHost().getHostAddress()}
+            : new String[] {
+              InetAddress.getLocalHost().getHostName(),
+              InetAddress.getLocalHost().getHostAddress(),
+              "localhost"
+            }) {
       ProcessBuilder builder = new ProcessBuilder();
       String[] psqlCommand =
           new String[] {
             "psql",
             String.format(
                 "sslmode=verify-full sslrootcert=%s host=%s port=%d",
-                publicKeyFile, InetAddress.getLocalHost().getHostName(), pgServer.getLocalPort()),
+                publicKeyFile, host, pgServer.getLocalPort()),
             "-c",
             "SELECT 1;"
           };
@@ -295,7 +330,8 @@ public class SSLMockServerTest extends AbstractMockServerTest {
 
       // Ignore known errors on some Macs on GitHub Actions.
       if (!(System.getProperty("os.name").contains("Mac")
-          && errors.contains("error: could not translate host name \""))) {
+          && (errors.contains("error: could not translate host name \"")
+              || errors.contains("does not match host name")))) {
         assertEquals(host, "", errors);
         assertEquals(" C \n---\n 1\n(1 row)\n", output);
         int res = process.waitFor();

